pkg/generate/code/check.go

```diff
@@ -0,0 +1,152 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package code
+
+import (
+	"fmt"
+	"strings"
+
+	awssdkmodel "github.com/aws/aws-sdk-go/private/model/api"
+
+	ackgenconfig "github.com/aws/aws-controllers-k8s/pkg/generate/config"
+	"github.com/aws/aws-controllers-k8s/pkg/model"
+	"github.com/aws/aws-controllers-k8s/pkg/names"
+)
+
+// CheckExceptionMessagePrefix returns Go code that contains a condition to
+// check if the message_prefix specified for a particular HTTP status code in
+// generator config is a prefix for the exception message returned by AWS API.
+// If message_prefix field was not specified for this HTTP code in generator
+// config, we return an empty string
+//
+// Sample Output:
+//
+// && strings.HasPrefix(awsErr.Message(), "Could not find model")
+func CheckExceptionMessagePrefix(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+	httpStatusCode int,
+) string {
+	rConfig, ok := cfg.ResourceConfig(r.Names.Original)
+	if ok && rConfig.Exceptions != nil {
+		excConfig, ok := rConfig.Exceptions.Errors[httpStatusCode]
+		if ok && excConfig.MessagePrefix != nil {
+			return fmt.Sprintf("&& strings.HasPrefix(awsErr.Message(), \"%s\") ",
+				*excConfig.MessagePrefix)
+		}
+	}
+	return ""
+}
+
+// CheckRequiredFieldsMissingFromShape returns Go code that contains a
+// condition checking that the required fields in the supplied Shape have a
+// non-nil value in the corresponding CR's Spec or Status substruct.
+//
+// Sample Output:
+//
+// return r.ko.Spec.APIID == nil || r.ko.Status.RouteID != nil
```

Comment:
correction in comment: This should be `return r.ko.Spec.APIID == nil || r.ko.Status.RouteID == nil`

Reply:
fixed.